docs/admin/authentication.md

```diff
@@ -52,6 +51,22 @@ and is a csv file with 3 columns: token, user name, user uid.
 When using token authentication from an http client the apiserver expects an `Authorization`
 header with a value of `Bearer SOMETOKEN`.
 
+**OpenID Connect ID Token** is enabled by passing the following options to the apiserver:
+- `--oidc-issuer-url` (required) tells the apiserver where to connect to the OpenID provider.
+- `--oidc-client-id` (required) is used by apiserver to verify the audience of the token.
+A valid [ID token](http://openid.net/specs/openid-connect-core-1_0.html#IDToken) MUST have this
+client-id in its `aud` claims.
+- `--oidc-ca-file` (optional) is used by apiserver to establish and verify the secure connection
+to the OpenID provider.
+
+Currently, the ID token will be obtained by some third-party app. This means the app and apiserver
+MUST share the `--oidc-client-id`. We expect `kubectl` to be able to get the token soon, as well as
+to refresh the token with a [refresh token](http://openid.net/specs/openid-connect-core-1_0.html#RefreshTokens)
+automatically when the token expires.
```

Comment:
Making kubectl talk to arbitrary identity servers feels like it might be straying a bit. I think I'd rather eventually see OAuth capabilities added to kubernetes in a way that kubectl could work against a known implementation, and the kubernetes server implementation could facilitate obtaining the OIDC token from the third-party.